src/lib.rs

```diff
@@ -253,35 +253,39 @@ fn expand_to_file(
         .truncate(true)
         .open(dest.as_path())?;
 
-    let Ok(mut f) = file_guard::try_lock(f.file_mut(), file_guard::Lock::Exclusive, 0, 64) else {
-        // the digest of the file will not match if the content to be written differed, hence any existing lock
-        // means we are already writing the same content to the file
-        if verbose {
-            eprintln!("expander: already in progress of writing identical content to {} by a different crate", dest.display());
-        }
-        // now actually wait until the write is complete
-        let _lock = file_guard::lock(f.file_mut(), file_guard::Lock::Exclusive, 0, 64)
-            .expect("File Lock never fails us. qed");
+    // Force implicit drop of FileGuard before rustfmt to prevent intermittent partial file locking issues on Windows (os error 33)
+    {
```

Comment:
I'd prefer a `drop(f)` with a comment, it avoids an indentation level.

Reply:
I've made the requested changes. This PR currently includes the changes from PR #14 to demonstrate the complete fix. Let me know if you want to merge PR #14 and do your own Windows testing, in which case I can rebase this PR to contain only the file guard changes, or if you're happy just to merge this one. As mentioned under PR14, I've retested it after the refactor.

Reply:
I currently do not have means to do windows targeted testing.

Since you already split them up, I'll merge #14 first and then rebase this one.

Reply:
Please don't merge yet, as per my previous comment.